firewood/src/manager.rs

```diff
@@ -339,14 +353,29 @@ impl RevisionManager {
     }
 
     pub fn revision(&self, root_hash: HashKey) -> Result<CommittedRevision, RevisionManagerError> {
-        self.by_hash
+        if let Some(revision) = self
```

Comment:
This multi-step flow could use some block level comments.

```suggestion
        // First, check the in-memory revision manager
        if let Some(revision) = self
```